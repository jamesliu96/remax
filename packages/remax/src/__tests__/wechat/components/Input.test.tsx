import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Input from '../../../adapters/wechat/components/Input';

describe('Input', () => {
  it('render correctly', () => {
    const tree = TestRenderer.create(<Input value="" type="text" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('render placeholderStyle as inline style', () => {
    const color = '#ffffff';

    const component: TestRenderer.ReactTestRenderer = TestRenderer.create(
      <Input placeholderStyle={{ color }} />
    );

    const instance = component.root.findByType('input');

    expect(instance.props['placeholder-style']).toEqual(`color:${color};`);
  });

  it('focus correctly', () => {
    const component: TestRenderer.ReactTestRenderer = TestRenderer.create(
      <Input />
    );

    const instance = component.root.findByType('input');

    expect(instance.props.focus).not.toBeTruthy();

    TestRenderer.act(() => {
      instance.props.bindtap();
    });

    expect(instance.props.focus).toBeTruthy();

    TestRenderer.act(() => {
      instance.props.bindblur();
    });

    expect(instance.props.focus).not.toBeTruthy();
  });
});
