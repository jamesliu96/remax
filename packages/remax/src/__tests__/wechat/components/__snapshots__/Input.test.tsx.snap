// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render correctly 1`] = `
<input
  bindblur={[Function]}
  bindfocus={[Function]}
  bindinput={[Function]}
  bindtap={[Function]}
  focus={false}
  maxlength={140}
  selection-end={999}
  selection-start={999}
  type="text"
  value=""
/>
`;
